contrib/storage-hive/core/src/test/java/org/apache/drill/exec/hive/complex_types/TestHiveStructs.java

```diff
@@ -0,0 +1,357 @@
+import org.apache.drill.test.TestBuilder;
+import org.apache.hadoop.hive.ql.Driver;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+
+import static java.util.Arrays.asList;
+import static org.apache.drill.exec.expr.fn.impl.DateUtility.parseBest;
+import static org.apache.drill.exec.expr.fn.impl.DateUtility.parseLocalDate;
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.junit.Assert.assertThat;
+
+@Category({SlowTest.class, HiveStorageTest.class})
+public class TestHiveStructs extends ClusterTest {
+
+  private static final JsonStringHashMap<String, Object> STR_N0_ROW_1 = TestBuilder.mapOf(
+      "f_int", -3000, "f_string", new Text("AbbBBa"), "f_varchar", new Text("-c54g"), "f_char", new Text("Th"),
+      "f_tinyint", -128, "f_smallint", -32768, "f_decimal", new BigDecimal("375098.406"), "f_boolean", true,
+      "f_bigint", -9223372036854775808L, "f_float", -32.058f, "f_double", -13.241563769628,
+      "f_date", parseLocalDate("2018-10-21"),
+      "f_timestamp", parseBest("2018-10-21 04:51:36"));
+
+  private static final JsonStringHashMap<String, Object> STR_N0_ROW_2 = TestBuilder.mapOf(
+      "f_int", 33000, "f_string", new Text("ZzZzZz"), "f_varchar", new Text("-+-+1"), "f_char", new Text("hh"),
+      "f_tinyint", 127, "f_smallint", 32767, "f_decimal", new BigDecimal("500.500"), "f_boolean", true,
+      "f_bigint", 798798798798798799L, "f_float", 102.058f, "f_double", 111.241563769628,
+      "f_date", parseLocalDate("2019-10-21"),
+      "f_timestamp", parseBest("2019-10-21 05:51:31"));
+
+  private static final JsonStringHashMap<String, Object> STR_N0_ROW_3 = TestBuilder.mapOf(
+      "f_int", 9199, "f_string", new Text("z x cz"), "f_varchar", new Text(")(*1`"), "f_char", new Text("za"),
+      "f_tinyint", 57, "f_smallint", 1010, "f_decimal", new BigDecimal("2.302"), "f_boolean", false,
+      "f_bigint", 101010L, "f_float", 12.2001f, "f_double", 1.000000000001,
+      "f_date", parseLocalDate("2010-01-01"),
+      "f_timestamp", parseBest("2000-02-02 01:10:09"));
+
+  private static final JsonStringHashMap<String, Object> STR_N2_ROW_1 = TestBuilder.mapOf("a",
+      TestBuilder.mapOf("b", TestBuilder.mapOf("c", 1000, "k", "Z")));
+
+  private static final JsonStringHashMap<String, Object> STR_N2_ROW_2 = TestBuilder.mapOf(
+      "a", TestBuilder.mapOf("b", TestBuilder.mapOf("c", 2000, "k", "X")));
+
+  private static final JsonStringHashMap<String, Object> STR_N2_ROW_3 = TestBuilder.mapOf(
+      "a", TestBuilder.mapOf("b", TestBuilder.mapOf("c", 3000, "k", "C")));
+
+  private static HiveTestFixture hiveTestFixture;
+
+  @BeforeClass
+  public static void setUp() throws Exception {
+    startCluster(ClusterFixture.builder(dirTestWatcher)
+        .sessionOption(ExecConstants.HIVE_OPTIMIZE_PARQUET_SCAN_WITH_NATIVE_READER, true)
+        .configProperty("drill.exec.rpc.user.timeout", 9999999)
```

Comment:
Could you please explain the reason for this change?

Reply:
I used option for debugging, just forgot to remove the line:)